pygeoapi/provider/hateoas.py

```diff
@@ -0,0 +1,205 @@
+                    jsondata = _get_json_data('{}/{}.json'.format(data_path, filename))
+                    resource_type = 'Assets'
+                except Exception as err:
+                    msg = 'Resource does not exist: {}'.format(data_path)
+                    LOGGER.error(msg)
+                    raise ProviderNotFoundError(msg)
+
+
+        if resource_type == 'Catalog' or resource_type == 'Collection':
+            content['type'] = resource_type
+
+            link_href_list = []
+            for link in jsondata["links"]:
+                if (resource_type == 'Catalog' and link["rel"] in ["child", "item"]) or \
+                   (resource_type == 'Collection' and link["rel"] in ["child", "item"]):
+                    link_href_list.append(link["href"].replace('\\', '/'))
+            link_href_list.sort()
+
+            for link in link_href_list:
+                unused, path_ending, entry_type = link.split('/')
+                newpath = os.path.join(baseurl, urlpath, path_ending).replace('\\', '/')
+
+                if entry_type == 'catalog.json':
+                    child_links.append({
+                        'rel': 'child',
+                        'href': newpath,
+                        'type': 'text/html',
+                        'created': "-",
+                        'entry:type': 'Catalog'
+                    })
+                elif entry_type == 'collection.json':
+                    child_links.append({
+                        'rel': 'child',
+                        'href': newpath,
+                        'type': 'text/html',
+                        'created': "-",
+                        'entry:type': 'Collection'
+                    })
+                else:
+                    child_links.append({
+                        'rel': 'item',
+                        'href': newpath,
+                        'title': path_ending,
+                        'created': "-",
+                        'entry:type': 'Item'
+                    })
+
+        elif resource_type == 'Assets':
+            content = jsondata
+            content['assets']['default'] = {
+                'href': os.path.join(baseurl, urlpath).replace('\\', '/'),
+            }
+
+            for key in content['assets']:
+                content['assets'][key]['file:size'] = 0
+                content['assets'][key]['created'] = jsondata["properties"]["datetime"]
+
+        content['links'].extend(child_links)
+
+        return content
+
+    def __repr__(self):
+        return '<HateoasProvider> {}'.format(self.data)
+
+def _get_json_data(jsonpath):
+    """
+    Helper function used to load a json file that is located on the WEB (HTTP request)
+    or on the server file system
+
+    :param jsonpath: path to the json file
+
+    :returns: `dict` of JSON item
+    """
+
+    if jsonpath[0:4].upper() == 'HTTP':
+        jsondata = requests.get(jsonpath).json()
+    else:
+        jsondata = json.load(open(jsonpath))
```

Comment:
Should this be wrapped in a context manager, i.e.:

```python
    else:
        with open(jsonpath) as fh:
            jsondata = json.load(fh)
```
```

Reply:
If you prefer this syntax, I can change my code to yours.

Reply:
Thanks; this is the preferred codepath to reading/writing files.